bindings/python/pymongoarrow/lib.pyx

```diff
@@ -59,11 +60,14 @@ def process_bson_stream(bson_stream, context):
     cdef bson_reader_t* stream_reader = bson_reader_new_from_data(docstream, length)
     cdef const bson_t * doc = NULL
     cdef bson_iter_t doc_iter
+    cdef bson_decimal128_t dec128
     cdef const char* key
     cdef bson_type_t value_t
     cdef Py_ssize_t count = 0
     cdef const char * bson_str
     cdef uint32_t str_len
+    cdef char *decimal128_str = <char *> malloc(
```

Comment:
Does this need to be a malloc or can it reside on the stack?

Reply:
It has to be a malloc unless we give it a literal number for the length.

Reply:
BSON_DECIMAL128_STRING is a literal number though. See http://mongoc.org/libbson/current/bson_decimal128_t.html#example

Can we do:
```
cdef char decimal128_str[BSON_DECIMAL128_STRING];
```

Reply:
I had tried that originally:

```
cdef char decimal128_str[BSON_DECIMAL128_STRING];
                                  ^
      ------------------------------------------------------------

      pymongoarrow/lib.pyx:69:29: Not allowed in a constant expression
```

Reply:
Oh, that's odd. One last question, does using a compile time redefinition work? EG:
```
DEF DEC128_STR_LEN = BSON_DECIMAL128_STRING
...
    cdef char decimal128_str[BSON_DECIMAL128_STRING];
```

https://cython.readthedocs.io/en/latest/src/userguide/language_basics.html#compile-time-definitions